eodag/plugins/download/aws.py

```diff
@@ -438,77 +598,196 @@ def download(
+                "complementary_url_key %s is missing in %s properties"
+                % (complementary_url_key, product.properties["id"])
+            )
 
-        if build_safe:
-            self.check_manifest_file_list(product_local_path)
+        # authenticate
+        authenticated_objects, s3_objects = self._do_authentication(
+            bucket_names_and_prefixes, auth
+        )
 
-        if asset_filter is None:
-            # save hash/record file
-            with open(record_filename, "w") as fh:
-                fh.write(product.remote_location)
-            logger.debug("Download recorded in %s", record_filename)
+        # downloadable files
+        unique_product_chunks = self._get_unique_products(
+            bucket_names_and_prefixes,
+            authenticated_objects,
+            asset_filter,
+            ignore_assets,
+            product,
+        )
+        assets_values = product.assets.get_values(asset_filter)
+        chunks_tuples = self._stream_download(
+            unique_product_chunks, product, build_safe, progress_callback, assets_values
+        )
+        outputs_filename = (
+            sanitize(product.properties["title"])
+            if "title" in product.properties
+            else sanitize(product.properties.get("id", "download"))
+        )
 
-            product.location = path_to_uri(product_local_path)
+        if len(assets_values) == 1:
+            first_chunks_tuple = next(chunks_tuples)
+            # update headers
+            filename = os.path.basename(list(unique_product_chunks)[0].key)
+            headers = {"content-disposition": f"attachment; filename={filename}"}
+            if assets_values[0].get("type", None):
+                headers["content-type"] = assets_values[0]["type"]
+
+            return dict(
+                content=chain(iter([first_chunks_tuple]), chunks_tuples),
+                headers=headers,
+            )
+        return dict(
+            content=stream_zip(chunks_tuples),
+            media_type="application/zip",
+            headers={
+                "content-disposition": f"attachment; filename={outputs_filename}.zip",
+            },
+        )
 
-        return product_local_path
+    def _stream_download(
+        self,
+        unique_product_chunks: Set[Any],
+        product: EOProduct,
+        build_safe: bool,
+        progress_callback: ProgressCallback,
+        assets_values: List[Dict[str, Any]],
+    ) -> Iterator[Tuple[str, datetime, int, Any, Iterator[Any]]]:
+        def yield_parts(bodies: List[Any], progress_callback: ProgressCallback):
+            for body in bodies:
+                for b in body:
+                    progress_callback(len(b))
+                    yield b
+
+        modified_at = datetime.now()
+        perms = 0o600
+        chunk_size = 4096 * 1024
+
+        for product_chunk in unique_product_chunks:
+            try:
+                chunk_rel_path = self.get_chunk_dest_path(
```

Comment:
please remove common path (using `os.path.commonpath`) from `chunk_rel_path` if `flatten_top_dirs` is `True`